### DISCUSSION TITLE
I don't understand the concurrency limits

### DISCUSSION THREAD

**Discussion by bpuchala** [cirruslabs/cirrus-ci-docs#1216] [Q&A]
I have multiple repositories where I'm trying to run two cibuildwheel tasks: 1 for Linux aarch64, and 1 for MacOS arm64. I'm receiving "Scheduling was delayed due to a concurrency limit on community tasks" messages sometimes, but not understanding exactly when I get them or what is supposed to happen next.

This build (and others from the same repository) ran two tasks concurrently:
- https://cirrus-ci.com/build/5677046676848640

These builds (and others from these repositories) are the same type of tasks, but get a "Scheduling was delayed..." message for either 1 or both tasks:
- https://cirrus-ci.com/build/6331102332190720
- https://cirrus-ci.com/build/5527818424025088

The message makes me think the jobs will run in sequence, but once I get the "Scheduling was delayed..." message for a task it doesn't seem to run even when another running task finishes. 

Are "delayed" tasks supposed to run eventually? Do I need to change the tasks? Or pay?

**Comment by fkorotkov** [comment]
Hey @bpuchala, sorry for the confusion. Your understanding is correct that tasks are getting in sequential queue upon reaching a per-platform limits (there are separate limits per macOS, Linux, etc.).

Unfortunately, there was another reason for the extra delays that you experienced. There was an issue with Arm Linux Containers around the time you run your builds that was contributing to the delays. The issue was happening between `5 PM EST on Aug 9, 2023` and `3 PM on Aug 10, 2023`.
